File: stralnumcmp_test.lua
require 'stralnumcmp';

assert(stralnumcmp('10km', '9km') > 0)
assert(stralnumcmp('part10.mp4', 'part9.mp4') > 0)
assert(stralnumcmp('9km', '10km') < 0)
assert(stralnumcmp('part9.mp4', 'part10.mp4') < 0)


-- just a ascii sort
assert(stralnumcmp("JPN-Z",    "JPN-A") > 0)
-- just a ascii sort
assert(stralnumcmp("2",        "1") > 0)
assert(stralnumcmp("vic1",     "vic0") > 0)
assert(stralnumcmp("vic10",    "vic9") > 0)
assert(stralnumcmp("10",       "9") > 0)
assert(stralnumcmp("E10",      "E9") > 0)
assert(stralnumcmp("10E",      "9E") > 0)
assert(stralnumcmp("1.10",     "1.9") > 0)
assert(stralnumcmp("aa11",     "11bba") > 0)
assert(stralnumcmp("aaaa",     "aaaa") == 0)
-- not a octal
assert(stralnumcmp("010",      "8") > 0)
-- longer is bigger
assert(stralnumcmp("10000",    "010000") < 0)
-- longer is bigger
assert(stralnumcmp("010000",   "10000") > 0)

File: stralnumcmp_test.c
/*
 gcc -Wall -O0 -o stralnumcmp_test stralnumcmp.c stralnumcmp_test.c
 */
#include <stdlib.h>
#include <stdio.h>

#include "stralnumcmp.h"

static int failed = 0;

#define ASSERT(_cond)							\
	do {								\
		if (!(_cond)) {						\
			fprintf(stderr, "ASSERT(%s) failed\n",		\
			    #_cond);					\
			failed++;					\
		}							\
	} while (0 /* CONSTCOND */)

int
main(int argc, char *argv[])
{
	ASSERT(stralnumcmp("JPN-Z",	"JPN-A") > 0);	/* just a ascii sort */
	ASSERT(stralnumcmp("2",	"1") > 0);	/* just a ascii sort */
	ASSERT(stralnumcmp("vic1",	"vic0") > 0);
	ASSERT(stralnumcmp("vic10",	"vic9") > 0);
	ASSERT(stralnumcmp("10",	"9") > 0);
	ASSERT(stralnumcmp("E10",	"E9") > 0);
	ASSERT(stralnumcmp("10E",	"9E") > 0);
	ASSERT(stralnumcmp("1.10",	"1.9") > 0);
	ASSERT(stralnumcmp("aa11",	"11bba") > 0);
	ASSERT(stralnumcmp("aaaa",	"aaaa") == 0);
	ASSERT(stralnumcmp("010",	"8") > 0);	/* not a octal */
	ASSERT(stralnumcmp("10000",	"010000") < 0);	/* longer is bigger */
	ASSERT(stralnumcmp("010000",	"10000") > 0);	/* longer is bigger */
	ASSERT(stralnumcmp("!11",	"aa") < 0);
	ASSERT(stralnumcmp("!11",	"11") < 0);

	if (failed != 0)
		abort();
	exit(EXIT_SUCCESS);
}
